File: testkit/src/main/scala/akka/kafka/testkit/javadsl/EmbeddedKafkaTest.scala
/*
 * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
 * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
 */

package akka.kafka.testkit.javadsl

import net.manub.embeddedkafka.{EmbeddedKafka, EmbeddedKafkaConfig}
import org.junit.jupiter.api.{AfterAll, BeforeAll}

/**
 * JUnit 5 aka Jupiter base-class with some convenience for creating an embedded Kafka broker
 * before running the tests.
 * Extending classes must be annotated with `@TestInstance(Lifecycle.PER_CLASS)` to create
 * a single instance of the test class with `@BeforeAll` and `@AfterAll` annotated methods called
 * by the test framework.
 */
abstract class EmbeddedKafkaTest extends KafkaTest {
  import EmbeddedKafkaJunit4Test._

  def kafkaPort: Int
  def replicationFactor = 1

  @BeforeAll def setupEmbeddedKafka() = startEmbeddedKafka(kafkaPort, replicationFactor)

  @AfterAll def cleanUpEmbeddedKafka() =
    stopEmbeddedKafka()
}

object EmbeddedKafkaTest {
  private def embeddedKafkaConfig(kafkaPort: Int, zooKeeperPort: Int, replicationFactor: Int) =
    EmbeddedKafkaConfig(kafkaPort,
                        zooKeeperPort,
                        Map(
                          "offsets.topic.replication.factor" -> s"$replicationFactor"
                        ))

  def startEmbeddedKafka(kafkaPort: Int, replicationFactor: Int): Unit =
    EmbeddedKafka.start()(embeddedKafkaConfig(kafkaPort, kafkaPort + 1, replicationFactor))

  def stopEmbeddedKafka(): Unit =
    EmbeddedKafka.stop()
}

File: testkit/src/main/java/akka/kafka/testkit/javadsl/EmbeddedKafkaTest.java
/*
 * Copyright (C) 2014 - 2016 Softwaremill <https://softwaremill.com>
 * Copyright (C) 2016 - 2020 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.kafka.testkit.javadsl;

import akka.actor.ActorSystem;
import net.manub.embeddedkafka.EmbeddedKafka$;
import net.manub.embeddedkafka.EmbeddedKafkaConfig;
import net.manub.embeddedkafka.EmbeddedKafkaConfig$;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import scala.collection.immutable.HashMap$;

/**
 * JUnit 5 aka Jupiter base-class with some convenience for creating an embedded Kafka broker before
 * running the tests. Extending classes must be annotated with `@TestInstance(Lifecycle.PER_CLASS)`
 * to create a single instance of the test class with `@BeforeAll` and `@AfterAll` annotated methods
 * called by the test framework.
 *
 * @deprecated Use testcontainers instead. Since 2.0.4.
 */
@Deprecated
public abstract class EmbeddedKafkaTest extends KafkaTest {

  private static EmbeddedKafkaConfig embeddedKafkaConfig(
      int kafkaPort, int zookeeperPort, int replicationFactor) {
    return EmbeddedKafkaConfig$.MODULE$.apply(
        kafkaPort,
        zookeeperPort,
        createReplicationFactorBrokerProps(replicationFactor),
        HashMap$.MODULE$.empty(),
        HashMap$.MODULE$.empty());
  }

  protected final int kafkaPort;
  protected final int replicationFactor;

  protected EmbeddedKafkaTest(ActorSystem system, int kafkaPort, int replicationFactor) {
    super(system, "localhost:" + kafkaPort);
    this.kafkaPort = kafkaPort;
    this.replicationFactor = replicationFactor;
  }

  protected EmbeddedKafkaTest(ActorSystem system, int kafkaPort) {
    this(system, kafkaPort, 1);
  }

  protected void startEmbeddedKafka(int kafkaPort, int replicationFactor) {
    EmbeddedKafka$.MODULE$.start(embeddedKafkaConfig(kafkaPort, kafkaPort + 1, replicationFactor));
  }

  protected void stopEmbeddedKafka() {
    EmbeddedKafka$.MODULE$.stop();
  }

  @BeforeAll
  void setupEmbeddedKafka() {
    startEmbeddedKafka(kafkaPort, replicationFactor);
  }

  @AfterAll
  void stopEmbeddedKafkaNow() {
    stopEmbeddedKafka();
  }
}
